pkg/converters/ingress/annotations/global_test.go

```diff
@@ -292,6 +292,36 @@ proxy_1
 	}
 }
 
+func TestCustomConfigFrontendLegacy(t *testing.T) {
+	testCases := []struct {
+		config   string
+		expected []string
+		logging  string
+	}{
```

Comment:
Thanks for this new test! It can be named "TestCustomConfigFrontendLate()" instead and test both options:

New field:

```diff
	testCases := []struct {
		config   string
+		legacy   string
		expected []string
		logging  string
	}{
```

New test cases:

```go
		// 2
		{
			legacy:   "# Legacy frontend\nhttp-response set-header X-Server HAProxy",
			expected: []string{"# Legacy frontend", "http-response set-header X-Server HAProxy"},
		},
		// 3
		{
			config:   "# New Header\nhttp-response set-header X-Server HAProxy",
			legacy:   "# Legacy frontend\nhttp-response set-header X-Server HAProxy",
			expected: []string{"# New Header", "http-response set-header X-Server HAProxy"},
			logging:  `WARN both config-frontend and config-frontend-late were used, ignoring config-frontend`,
		},
```

New global data initialization:

```go
		d := c.createGlobalData(map[string]string{
			ingtypes.GlobalConfigFrontend:     test.legacy,
			ingtypes.GlobalConfigFrontendLate: test.config,
		})
```

Reply:
indeed, thanks